substrate/frame/bounties/src/lib.rs

```diff
@@ -112,46 +116,87 @@ pub use weights::WeightInfo;
 pub use pallet::*;
 
 type BalanceOf<T, I = ()> = pallet_treasury::BalanceOf<T, I>;
-
-type PositiveImbalanceOf<T, I = ()> = pallet_treasury::PositiveImbalanceOf<T, I>;
+type AssetBalanceOf<T, I = ()> = pallet_treasury::AssetBalanceOf<T, I>;
+type BeneficiaryLookupOf<T, I = ()> = pallet_treasury::BeneficiaryLookupOf<T, I>;
 
 /// An index of a bounty. Just a `u32`.
 pub type BountyIndex = u32;
 
 type AccountIdLookupOf<T> = <<T as frame_system::Config>::Lookup as StaticLookup>::Source;
 
+type BountyType<T, I> = Bounty<
+	<T as frame_system::Config>::AccountId,
+	BalanceOf<T, I>,
+	AssetBalanceOf<T, I>,
+	BlockNumberFor<T>,
+	<T as pallet_treasury::Config<I>>::AssetKind,
+	<<T as pallet_treasury::Config<I>>::Paymaster as Pay>::Id,
+	<T as pallet_treasury::Config<I>>::Beneficiary,
+>;
+
 /// A bounty proposal.
 #[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
-pub struct Bounty<AccountId, Balance, BlockNumber> {
+pub struct Bounty<
+	AccountId,
+	NativeBalance,
+	AssetBalance,
+	BlockNumber,
+	AssetKind,
+	PaymentId,
+	Beneficiary,
+> where
+	Beneficiary: Clone,
+	PaymentId: Clone,
+{
 	/// The account proposing it.
 	proposer: AccountId,
-	/// The (total) amount that should be paid if the bounty is rewarded.
-	value: Balance,
-	/// The curator fee. Included in value.
-	fee: Balance,
+	// TODO: new filed, migration required.
+	/// The kind of asset this bounty is rewarded in.
+	asset_kind: AssetKind,
+	/// The (total) amount of the `asset_kind` that should be paid if the bounty is rewarded.
+	value: AssetBalance,
```

Comment:
I would keep this just `Balance`, since there wont be other balance types. Below for deposit and bond we will have consideration ticket type.